tests/contest/runtimetest/src/tests.rs

```diff
@@ -550,6 +552,44 @@ pub fn test_io_priority_class(spec: &Spec, io_priority_class: IOPriorityClass) {
     }
 }
 
+pub fn test_validate_root_readonly(spec: &Spec) {
+    let root = spec.root().as_ref().unwrap();
+    if root.readonly().unwrap() {
+        if let Err(e) = test_dir_write_access("/") {
+            let errno = Errno::from_raw(e.raw_os_error().unwrap());
+            if errno == Errno::EROFS {
+                /* This is expected */
+            } else {
+                eprintln!(
+                    "readonly root filesystem, error in testing write access for path /, error: {}",
+                    errno
+                );
+            }
+        }
+        if let Err(e) = test_dir_read_access("/") {
+            let errno = Errno::from_raw(e.raw_os_error().unwrap());
+            if errno == Errno::EROFS {
+                /* This is expected */
+            } else {
+                eprintln!(
+                    "readonly root filesystem, error in testing read access for path /, error: {}",
+                    errno
+                );
+            }
+        }
+    } else if let Err(e) = test_dir_write_access("/") {
+        if e.raw_os_error().is_some() {
+            let errno = Errno::from_raw(e.raw_os_error().unwrap());
+            eprintln!(
+                "readt only root filesystem is false but write access for path / is err, error: {}",
+                errno
+            );
+        } else {
+            /* This is expected */
+        }
+    }
```

Comment:
Why we are checking `raw_os_error().is_some()` ? If this was error, any error while writing, it should be considered as a test fail, right? Also can we have a read_access check in this branch as well? Basically :

if readonly=true{
   check write_access == error;
   check read_access == ok;

}else{
   check write_access == ok;
   check read_access == ok;
}